qualtran/_infra/data_types.py

```diff
@@ -528,6 +581,13 @@ def from_bits(self, bits: Sequence[int]) -> Fxp:
         fxp_bin = "0b" + bits_bin[: -self.num_frac] + "." + bits_bin[-self.num_frac :]
         return Fxp(fxp_bin, dtype=self.fxp_dtype_str)
 
+    def from_bits_array(self, bits_array: NDArray[np.uint8]):
+        assert isinstance(self.bitsize, int), "cannot convert to bits for symbolic bitsize"
+        # TODO figure out why `np.vectorize` is not working here
```

Comment:
open an issue and link? Do you have any theories? as I understand it: `np.vectorize` just does a python for loop under-the-hood

Reply:
I think it's something to do with how Fxp interacts with numpy. Fxp has some inbuilt support to operate over NDArrays, so perhaps mixing the order up causes issues. I didn't investigate more though.

Reply:
An `Fxp` object can wrap a numpy array -- so to get a ND collection of Fxp objects, you construct a `Fxp(numpy_array_of_int_or_float_values)` instead of `np.array([Fxp(x) for x in array_of_int_or_float_values])`

See https://github.com/francof2a/fxpmath?tab=readme-ov-file#arithmetic for more details